src/detect-fast-pattern.c

```diff
@@ -282,10 +283,11 @@ static int DetectFastPatternSetup(DetectEngineCtx *de_ctx, Signature *s, const c
                        "for fast_pattern offset");
             goto error;
         }
-        int offset = atoi(arg_substr);
-        if (offset > 65535) {
-            SCLogError(SC_ERR_INVALID_SIGNATURE, "Fast pattern offset exceeds "
-                       "limit");
+        int16_t offset;
+        if (ByteExtractStringInt16(&offset, 10, 0,
```

Comment:
Should it rather be Uint16 ? 
Otherwise, it needs to be Int32 so that we can go up to 65535...

Reply:
Yes OK